contracts/rosca.sol

```diff
@@ -1,266 +1,258 @@
-
-
-    //added
-    event LogFundsWithdrawal(address user, uint amount,address destination);
-    event LogStartOfPeriod(uint currentRound);
-
-    //state variables
-    address foreman;
-    uint roundSum;
-    uint numRounds;
-    uint currentEpoch;
-    uint numEpochs;
-    int8 minParticipants;
-    uint roundPeriodInDays;
-    uint startTime;
-    int16 feeToService;
-    address feeAddress;
-    uint currentRound;  //currentRound will be set to 0 when ROSCA is created and will turn to one when the ROSCA actually start
-
-    /*
-        not sure about which structures to use yet
+contract ROSCA {
+  address constant WETRUST = 0x0;
+  uint constant MIN_CONTRIBUTION_SIZE = 1000000000000;
+  uint constant MAX_FEE_IN_THOUSANDTH = 20;
+
+  event LogParticipantRegistered(address user);
+  event LogContributionMade(address user, uint amount);
+  event LogNewLowestBid(uint bid,address winnerAddress);
+  event LogRoundFundsRelease(address winnerUniqName, uint amountInWei);
+  event LogFundsWithdrawal(address user, uint amount,address destination);
+  event LogStartOfPeriod(uint currentRound);
+
+  //state variables
+  uint16 numRoundsInEpoch;     //number of rounds in 1 epoch, user doesnt have control over this, it'll increment everytime a new participant is registered
+  uint16 currentEpoch;
+  uint16 numEpochs;     //number of Epochs this contract will run
+  uint16 roundPeriodInDays;
+  uint16 serviceFeeInThousandths;
+  uint16 currentRound;  //currentRound will be set to 0 when ROSCA is created and will turn to one when the ROSCA actually start
+  uint8 minParticipants;
+  address foreman;    //address of the foreman
+  address feeAddress;
+  uint contributionSize;
+  uint startTime;     //startTime must be in timestamp
+
+  struct USER{
+    uint contributed; // number of times contributed or could be amount contributed
+    uint paid; // number of times bid Won
+    uint pendingWithdrawl; // how much they are allowed to withdraw, i.e if someone win , their allowance will go up by the bid.
+    bool alive; //needed to check if a member is indeed a member
+  }
+  mapping(address => USER) members; //using struct USER to keep track of contributions and paid, allowance and etc.
+  address[] membersAddresses;    //this is the only way to iterate through all the member's address
+
+  mapping(address => bool) pendingJoinRequest; // this way , address can be used as index, if we use address[] , we'll have to go thru a whole array
+
+  //bidding related state variable
+  uint lowestBid;
+  address winnerAddress;
+
+
+  modifier onlyForeman {if(msg.sender != foreman) throw;_;}
+  modifier beforeStart {if(currentRound != 0)throw;_;}
+
+  /**
+    * Creates a new ROSCA and initialize the necessary variables, ROSCA doesnt start untill the specified startTime
+    * Creator of the contract becomes foreman and also added as the first member of the ROSCA
+    * params:
+    * - roundPeriodInDays uint16
```

Comment:
i think there is a way to specify param explanations in a structured way , so that tools can later use that:
https://github.com/ethereum/wiki/wiki/Ethereum-Natural-Specification-Format

Even otherwise, I suggest not mentioning the param type in the comments, as it may get out of sync with the actual param (when someone changes the type of the actual param)